# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by scripts/Create-CopyPatch.
# 
# T2 SDE: architecture/sparc/package/*/hotfix.patch
# Copyright (C) 2021 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

From 3f512ca79fda7854da188df0585e71467119206e Mon Sep 17 00:00:00 2001
From: Brett Neumeier <brett@neumeier.us>
Date: Mon, 31 Aug 2015 15:27:21 -0700
Subject: [PATCH] Fix non-v9 32-bit sparc build.

	[BZ #18870]
	* sysdeps/sparc/sparc32/sem_open.c: Add missing #include
diff --git a/sysdeps/sparc/sparc32/sem_open.c b/sysdeps/sparc/sparc32/sem_open.c
index 16cb9ad591..59df2d7b9b 100644
--- a/sysdeps/sparc/sparc32/sem_open.c
+++ b/sysdeps/sparc/sparc32/sem_open.c
@@ -29,6 +29,7 @@
 #include <sys/mman.h>
 #include <sys/stat.h>
 #include "semaphoreP.h"
+#include <futex-internal.h>
 #include <shm-directory.h>
 
 
-- 
2.27.0


https://sourceware.org/git/?p=glibc.git;a=commit;h=43c2948756bb6e144c7b871e827bba37d61ad3a3
https://sourceware.org/git/?p=glibc.git;a=commit;h=b87c1ec3fa398646f042a68f0ce0f7d09c1348c7

--- glibc-2.22/sysdeps/sparc/sparc32/fpu/e_sqrtl.c.vanilla	2021-02-09 16:17:34.563253198 +0100
+++ glibc-2.22/sysdeps/sparc/sparc32/fpu/e_sqrtl.c	2021-02-09 16:17:53.305254165 +0100
@@ -1 +1,28 @@
-/* __ieee754_sqrtl is defined in q_sqrt.c.  */
+/* Long double square root, sparc32 version.
+   Copyright (C) 2016 Free Software Foundation, Inc.
+   This file is part of the GNU C Library.
+
+   The GNU C Library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+
+   The GNU C Library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License along with the GNU C Library; if not, see
+   <http://www.gnu.org/licenses/>.  */
+
+#include <math.h>
+
+extern long double _Q_sqrt(const long double a);
+
+long double
+__ieee754_sqrtl (long double x)
+{
+  return _Q_sqrt (x);
+}
+strong_alias (__ieee754_sqrtl, __sqrtl_finite)
--- glibc-2.22/sysdeps/sparc/sparc32/soft-fp/q_sqrt.c.vanilla	2021-02-09 16:17:34.565253199 +0100
+++ glibc-2.22/sysdeps/sparc/sparc32/soft-fp/q_sqrt.c	2021-02-09 16:17:53.305254165 +0100
@@ -35,4 +35,3 @@
   FP_HANDLE_EXCEPTIONS;
   return c;
 }
-strong_alias (_Q_sqrt, __ieee754_sqrtl);
--- glibc-2.22/sysdeps/unix/sysv/linux/sparc/sparc32/libm.abilist.vanilla	2021-02-09 16:17:34.567253199 +0100
+++ glibc-2.22/sysdeps/unix/sysv/linux/sparc/sparc32/libm.abilist	2021-02-09 16:18:39.514256549 +0100
@@ -379,6 +379,7 @@
  __sinhl_finite F
  __sqrt_finite F
  __sqrtf_finite F
+ __sqrtl_finite F
  __y0_finite F
  __y0f_finite F
  __y0l_finite F
--- glibc-2.22/sysdeps/unix/sysv/linux/sparc/sparc32/vfork.S.vanilla	2021-02-09 16:21:21.813264922 +0100
+++ glibc-2.22/sysdeps/unix/sysv/linux/sparc/sparc32/vfork.S	2021-02-09 16:22:22.169268036 +0100
@@ -21,7 +21,7 @@
 
 	.text
 	.globl		__syscall_error
-ENTRY(__vfork)
+ENTRY(__libc_vfork)
 	ld	[%g7 + PID], %o5
 	cmp	%o5, 0
 	bne	1f
@@ -42,8 +42,10 @@
 	 st	%o5, [%g7 + PID]
 1:	retl
 	 nop
-END(__vfork)
+END(__libc_vfork)
 
+#if IS_IN (libc)
+weak_alias (__libc_vfork, vfork)
+strong_alias (__libc_vfork, __vfork)
 libc_hidden_def (__vfork)
-weak_alias (__vfork, vfork)
-strong_alias (__vfork, __libc_vfork)
+#endif
